### DISCUSSION TITLE
Remove unsupported target frameworks and optimize referenced packages

### DISCUSSION THREAD

**Discussion by d4n3436** [discord-net/Discord.Net#2294] [Ideas]
Currently there are 2 frameworks that Discord.Net targets and are now out of support:
- .NET Framework 4.6.1 (April 26, 2022)
- .NET 5 (May 08, 2022)

These 2 frameworks should be removed in a future version because they are now out of support and they're adding unnecessary bloat to the library.

Additionally, the inclusion of some packages can be optimized by only referencing them if they are required by the target framework:
- `System.Collections.Immutable` (included in .NET 6 but not in .NET Standard 2.x)
- `System.ValueType` (included in .NET Standard 2.x and .NET 6 so it can be removed)
- `System.Reactive` (unused, in `Discord.Net.Interactions`)
- `System.Interactive.Async` (in `Discord.Net.Core`, can be replaced with the package it references, `System.Linq.Async`)

**Comment by csmir** [comment]
I think this is something that should be put up for discussion instead. Reason behind why we should remove unsupported framework is clear, but there may be a large amount of people who disagree with this purely for legacy sake. I'd like to hear more feedback about this change.

**Comment by Almighty-Shogun** [comment]
For me I honestly don't care since it doesn't effect me. If removing it is better for the package then I say go for it.  But what @Rozen4334 says is correct.

> but there may be a large amount of people who disagree with this purely for legacy sake

Some people don't or can't use the newer .NET versions and are using the lower frameworks.

**Comment by SylveonDeko** [comment]
For optimizations sake do it

**Comment by Jelosus2** [comment]
Can be a good idea, but for people like me who don't have a good pc that can run vs 2022 smoothly will be a problem since vs 2019 don't support .NET 6, im using .NET 5 right now

**Reply by d4n3436** [reply]
Discord.Net would still target .NET Standard 2.0, so it would still support .NET Framework 4.6.1+ and .NET Core 2.0+.

**Reply by Jelosus2** [reply]
I never used Framework or Core

**Reply by Jelosus2** [reply]
Nvm, i won't have any problem, you can do the changes
